plugins/chat/assets/javascripts/discourse/routes/chat.js

```diff
@@ -57,6 +59,16 @@ export default class ChatRoute extends DiscourseRoute {
   }
 
   activate() {
+    withPluginApi("1.8.0", (api) => {
+      api.setSidebarPanel("chat");
+      if (getUserChatSeparateSidebarMode(this.currentUser).never) {
+        api.setCombinedSidebarMode();
+        api.hideSidebarSwitchPanelButtons();
+      } else {
+        api.setSeparatedSidebarMode();
+      }
+    });
```

Comment:
This if-else is also present twice, with a slight difference. Another copy is here:

https://github.com/discourse/discourse/blob/7ca178f99e4095dde8bde28e9939c1ab18d24c5f/plugins/chat/assets/javascripts/discourse/services/chat-state-manager.js#L62-L69

Not sure if this intentional and if we are able to get rid of it this time, but this is a duplication and ideally should be dried up.

Reply:
The condition is inverted, but looks like both instances do more or less the same.

Reply:
So yes this one is intentional, it's actually slightly different behavior for drawer, and for now at least I prefer the separation instead of a clunky abstraction. I expect this will keep diverging.